### ISSUE TITLE
Support project-wide configuration

### ISSUE THREAD

**Issue by atemate** [neuro-inc/neuro-cli#1931]
Use-case 1: to be able to work on 2 different projects targeting 2 different clusters.
Use-case 2: to be able to work being authenticated against 2 different users (for example, `<my-name>` and `ci-bot` -- needed for debugging purposes).
Use-case 3: since (1) neuro configuration is global (OS user-wide) but (2) `pip` can install `neuro` in multiple different ways (system global / OS user-wide / virtualenv-wide / conda env wide),  and at the same (3) time neuro changes its database format (stored in `~/.neuro) so quickly without backward compatibility, working simultaneously on multiple projects is impossible as neuro is constantly failing to read its authentication config.

**Comment by YevheniiSemendiak** [comment]
`kubectl` tool with its *contexts* notion looks handy for me.
In `kubectl` you could have a single config file with multiple contexts defined, each corresponds to the specific user+cluster pair.

We could think of mimicking it somehow.
In this case, we might allow to log-in for multiple users with one CLI instance and store their config separately.
Later you could do smth like `neuro config switch-user <needed-user`, which will dump the current user config back into local DB and load the target user config 🤔

**Comment by YevheniiSemendiak** [comment]
Turned out we already have such functionality, idk why I haven't found it previously https://github.com/neuro-inc/neuro-cli/blob/3231df329df0b3c28ad86c8edc835762bc7a4f4c/neuro-sdk/src/neuro_sdk/_config_factory.py#L38

tl; dr: use `NEUROMATION_CONFIG` env var to define a filesystem path, where CLI config should be stored (`~/.neuro` is the default).
This allows one to switch between the contexts (control plane, user, cluster) by changing this var, which is already good. I was thinking about somewhat similar UX to the `kubectl config use-context`, but this is mostly needed for our internal team, so let's just close this issue.

fyi, @anayden, @andriihomiak, @eviltwinhans
